packages/devtools/src/devtoolsdriver.ts

```diff
@@ -108,6 +94,51 @@ export default class DevToolsDriver {
         log.trace(`Switching to window handle with id ${pageIds[0]}`)
     }
 
+    private addListener(emitter: EventEmitter, eventName: string, handler: any) {
+        const boundHandler = handler.bind(this)
+        emitter.on(eventName, boundHandler)
+        this.activeListeners.push({ emitter, eventName, boundHandler })
+    }
+
+    private offAllListeners() {
+        this.activeListeners.forEach(({ emitter, eventName, boundHandler }) => {
+            emitter.off(eventName, boundHandler)
```

Comment:
do we need to run this when the listeners are removed anyway?

Reply:
We should unplug listeners from emitters in case a listener instance isn't destroyed together with emitter. Here pages and browsers emitters can be created and destroyed (new windows, session reload) but DevToolsDriver with its listeners remains all session. Otherwise the emitters won't be garbage-collected and we could have also unpredicted behavior (e.g. any event from hanging window), however the latter is less probable here.

Reply:
I wonder if we could automatically do this on an event so that it always happens and not just when explicitly calling `cleanupListeners` as it would be more robust and better for maintenance. Do you happen to know if this is possible?

Reply:
It could be probably possible to do it attaching some logic to `targetdestroyed` event handler, but it would require some more investigation/tests - probably out of scope of this PR.